pwndbg/commands/heap.py

```diff
@@ -23,37 +32,44 @@
 from pwndbg.heap.ptmalloc import Bins
 from pwndbg.heap.ptmalloc import BinType
 from pwndbg.heap.ptmalloc import Chunk
-from pwndbg.heap.ptmalloc import DebugSymsHeap
+from pwndbg.heap.ptmalloc import GlibcMemoryAllocator
 from pwndbg.heap.ptmalloc import Heap
 
 
-def read_chunk(addr):
+def read_chunk(addr: int) -> Dict[str, int]:
     """Read a chunk's metadata."""
     # In GLIBC versions <= 2.24 the `mchunk_[prev_]size` field was named `[prev_]size`.
     # To support both versions, change the new names to the old ones here so that
     # the rest of the code can deal with uniform names.
+    assert isinstance(pwndbg.heap.current, GlibcMemoryAllocator)
+    assert pwndbg.heap.current.malloc_chunk is not None
     renames = {
         "mchunk_size": "size",
         "mchunk_prev_size": "prev_size",
     }
-    if isinstance(pwndbg.heap.current, DebugSymsHeap):
-        val = pwndbg.gdblib.typeinfo.read_gdbvalue("struct malloc_chunk", addr)
+    if isinstance(pwndbg.heap.current.malloc_chunk, gdb.Type):
```

Comment:
This seems like different behavior to me, but I'm not very familiar with this code. @CptGibbon or @lebr0nli, can you verify this is correct?

Reply:
This is a two-in-one check. `pwndbg.heap.current.malloc_chunk` is only of type `gdb.Type` if the allocator is `DebugSymsHeap`. Otherwise it would be `pwndbg.heap.structs.CStruct2GDB`. I can explicitly keep the previous check for `DebugSymsHeap` too for clearity.

Reply:
Ah, this is from an older iteration of the ptmalloc typings where the properties like `malloc_chunk` had a union return type. Since they have the correct one now the old check for DebugSymsHeap is enough. I'll revert, good catch.

Reply:
Or were you commenting on the change of 
```python
val = pwndbg.gdblib.typeinfo.read_gdbvalue("struct malloc_chunk", addr)
```
to
```python
val = pwndbg.gdblib.memory.poi(pwndbg.heap.current.malloc_chunk, addr)
```

`memory.poi` was used in ptmalloc and it seemed better to not resolve the type again so I went with the `poi` way. Both lines do the same thing though.

https://github.com/pwndbg/pwndbg/blob/92640c7131409b01fb096b9ff796b63f3ddc8520/pwndbg/heap/ptmalloc.py#L157